spec/controllers/searchgov_urls_controller_spec.rb

```diff
@@ -1,26 +1,27 @@
 require 'spec_helper'
 
 describe Admin::SearchgovUrlsController do
-  # let!(:searchgov_domain) { SearchgovDomain.create(domain: 'agency.gov')}
+  fixtures :users
+
+  let(:admin) { users(:affiliate_admin) }
   let(:searchgov_url) { SearchgovUrl.create(url: 'agency.gov/test') }
   let(:params) do
     { association: 'searchgov_urls', parent_scaffold: 'admin/searchgov_domains',
     searchgov_domain_id: searchgov_url.searchgov_domain.id, id: searchgov_url.id }
   end
 
-  describe '#fetch' do
-    # before do
-    #   # post :fetch, association: 'searchgov_urls', parent_scaffold: 'admin/searchgov_domains',
-    #   # searchgov_domain_id: searchgov_url.searchgov_domain.id, id: searchgov_url.id
-    #   # post "/admin/searchgov_domains/#{searchgov_url.searchgov_domain.id}/searchgov_urls/#{searchgov_url.id}/fetch?_method=post&association=searchgov_urls&parent_scaffold=admin%2Fsearchgov_domains&searchgov_domain_id=#{searchgov_url.searchgov_domain.id}"
-    #   post :fetch, params
-    #   p response.body
-    # end
+  context 'when logged in as an admin' do
+    describe '#fetch' do
+      before do
+        activate_authlogic
+        UserSession.create admin
+        allow(SearchgovDomainPreparerJob).to receive(:perform_later)
+      end
 
-    it 'enqueues a searchgov_url_fetcher_job to the searchgov queue' do
-      # ActiveJob::Base.queue_adapter = :test
-      expect{ post :fetch, params }.
-        to have_enqueued_job(SearchgovUrlFetcherJob).on_queue("searchgov")
+      it 'enqueues a searchgov_url_fetcher_job to the searchgov queue' do
+        expect{ post :fetch, params }.
+          to have_enqueued_job(SearchgovUrlFetcherJob).on_queue("searchgov")
```

Comment:
@ethanewing , TDD is helpful for bug fixes as well. This test doesn't actually test this change, because it's not checking the job arguments: https://github.com/GSA/search-gov/commit/e7b3adb.

Reply:
Would that test to make sure it only takes named arguments go in `searchgov_url_fetcher_job_spec.rb` since that test is specifically about the functionality of the `perform` function within the model?  I've already got a test in that spec that tests that the `perform` is invalid if it isn't given an argument -- should I just add another that tests that it's invalid if it's not given a named argument?

Reply:
This test is covering the bug you fixed with https://github.com/GSA/search-gov/commit/e7b3adb. It ensures we're passing the args correctly to the job. You can do that by specifying the args in your test:
```ruby
expect{ post :fetch, params }.
  to have_enqueued_job(SearchgovUrlFetcherJob).on_queue("searchgov").
  with(searchgov_url: searchgov_url)
```

It couldn't hurt to also make your `searchgov_url_fetcher_job_spec.rb` spec more specific, by testing that it requires a`searchgov_url` keyword argument.